app/code/core/Mage/Usa/Model/Shipping/Carrier/Dhl/International.php

```diff
@@ -748,7 +748,7 @@ protected function _getDimension($dimension, $configWeightUnit = false)
         $countryDimensionUnit = $this->getCode('dimensions_variables', $this->_getDimensionUnit());
 
         if ($configDimensionUnit != $countryDimensionUnit) {
-            $dimension = Mage::helper('usa')->convertMeasureDimension(
+            $dimension = (float) Mage::helper('usa')->convertMeasureDimension(
```

Comment:
This is incorrectly fixed, needs to be cast to float inside of round() inside of the convertMeasureDimension() not outside of it. Will still error.
needs to be:
$dimension = Mage::helper('usa')->convertMeasureDimension(
    round((float)$dimension,3),
    $configDimensionUnit,
    $countryDimensionUnit
);

Reply:
I created PR https://github.com/OpenMage/magento-lts/pull/2193 to fix this.